### PR TITLE
Add chat HUD flag

### PR DESCRIPTION
**Pull request by GreenXenith**
Adds new HUD flag `chat` to disable/enable the chat on the HUD. This does not affect the console.

If chat is disabled by the server, for consistency, attempting to toggle chat will result in the standard "disabled by game or mod" message (as much as I dislike it).

The console should be separate from the HUD chat, as I may want to still use the console for inputs but not necessarily display it on the HUD. If we want a way to disable that too, different PR, different time.

Use cases:
* A game implements it's own chat interface
* Chat is not applicable to a game
* A game wants to hide as much HUD as it can during a wait period
* ~~A user needs to be disallowed to see chat~~

### CODE REVIEW

**Comment by sfan5** [comment]
> Use cases: A user needs to be disallowed to see chat

well this doesn't really help since as you said the chat console is still usable.

File: src/hud.h [hunk 1]
```diff
@@ -48,6 +48,7 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #define HUD_FLAG_MINIMAP_VISIBLE       (1 << 5)
 #define HUD_FLAG_MINIMAP_RADAR_VISIBLE (1 << 6)
 #define HUD_FLAG_BASIC_DEBUG           (1 << 7)
+#define HUD_FLAG_CHAT_VISIBLE          (1 << 8)
```
Comment (sfan5) [inline src/hud.h:51]: not `HUD_FLAG_CHAT`?

Comment (GreenXenith) [reply src/hud.h:51]: The rest use visible, keeping it consistent.

**Comment by GreenXenith** [comment]
> > Use cases: A user needs to be disallowed to see chat
> 
> well this doesn't really help since as you said the chat console is still usable.

Sorry yeah, fair enough. The rest still applies though.

**Review by sfan5 — APPROVED** [review]
Works.

**Review by SmallJoker — APPROVED** [review]
New client <--> Old server works well. No issues found.

**Pull request merged**